src/lib/telemetry/TelemetryReporter.ts

```diff
@@ -117,16 +139,39 @@ export class TelemetryReporter {
       }
       return error
     })
+    await ensureDir(join(TelemetryReporter.PENDING_DATA_DIR, 'errors'))
     await ensureFile(metaErrorFilePath)
-    await writeJson(metaErrorFilePath, { errors: errorsReport })
+    await writeJson(metaErrorFilePath, errorsReport)
   }
 
-  private async pendingFilesPaths() {
-    const pendingDataFiles = (await readdir(TelemetryReporter.PENDING_DATA_DIR)).filter(fileName => {
-      return fileName !== this.dataPendingLock.lockName
-    })
+  public async ensureTelemetryDirMazSize() {
+    const telemetryFiles = await readdirRecursive(TelemetryCollector.TELEMETRY_LOCAL_DIR)
+    let telemetryDirSize = 0
+    await Promise.all(telemetryFiles.map(async (file) => {
+      const fileStats = await stat(file)
+      telemetryDirSize += fileStats.size
+    }))
+    if (telemetryDirSize > 10 * 1000 * 1000) {
+      await remove(TelemetryCollector.TELEMETRY_LOCAL_DIR)
+    }
+  }
+
+  private async treatReportError(reportError, reportType: 'metrics' | 'errors', metricsOrErrors: MetricReport[] | ErrorReport[]) {
```

Comment:
Just a suggestion - typescript allows overloading function types (https://www.tutorialsteacher.com/typescript/function-overloading - https://www.typescriptlang.org/docs/handbook/functions.html#overloads), this way we can tell that `treatReportError` will accept:
`reportError:any, reportType: 'metrics', metrics: MetricReport[])` or
`reportError:any, reportType: 'error', errors: ErrorReport[])`
This way we get autocomplete the moment we write `reportType = 'error'` for example, because the `ts-server` will know that the only option for the third parameter is `ErrorReport[]`.

Without overloading we can write `reportType = 'error'` and still give to the third parameter a `MetricReport[]`

Reply:
Unfortunately though I think `eslint` doesn't understand overloads well yet, so it will complain about it, I would add a `// eslint-disable-next-line nameOfTheRule`